x-pack/plugins/task_manager/server/task_claimers/strategy_mget.ts

```diff
@@ -0,0 +1,366 @@
+        task.scheduledAt = task.retryAt;
+      } else {
+        task.scheduledAt = task.runAt;
+      }
+      task.retryAt = claimOwnershipUntil;
+      task.ownerId = taskStore.taskManagerId;
+      task.status = TaskStatus.Claiming;
+
+      return task;
+    });
+
+  const finalResults: ConcreteTaskInstance[] = [];
+  let conflicts = staleTasks.size;
+  let bulkErrors = 0;
+
+  try {
+    const updateResults = await taskStore.bulkUpdate(taskUpdates, { validate: false });
+    for (const updateResult of updateResults) {
+      if (isOk(updateResult)) {
+        finalResults.push(updateResult.value);
+      } else {
+        const { id, type, error } = updateResult.error;
+
+        // this check is needed so error will be typed correctly for isConflictError
+        if (SavedObjectsErrorHelpers.isSavedObjectsClientError(error)) {
+          if (SavedObjectsErrorHelpers.isConflictError(error)) {
+            conflicts++;
+          } else {
+            logger.warn(
+              `Saved Object error updating task ${id}:${type} during claim: ${error.error}`,
+              logMeta
+            );
+            bulkErrors++;
+          }
+        } else {
+          logger.warn(`Error updating task ${id}:${type} during claim: ${error.message}`, logMeta);
+          bulkErrors++;
+        }
+      }
+    }
+  } catch (err) {
+    logger.warn(`Error updating tasks during claim: ${err}`, logMeta);
+  }
+
+  // separate update for removed tasks; shouldn't happen often, so unlikely
+  // a performance concern, and keeps the rest of the logic simpler
+  let removedCount = 0;
+  if (removedTasks.size > 0) {
+    const tasksToRemove = Array.from(removedTasks);
+    for (const task of tasksToRemove) {
+      task.status = TaskStatus.Unrecognized;
+    }
+
+    // don't worry too much about errors, we'll get them next time
+    try {
+      const removeResults = await taskStore.bulkUpdate(tasksToRemove, { validate: false });
+      for (const removeResult of removeResults) {
+        if (isOk(removeResult)) {
+          removedCount++;
+        } else {
+          const { id, type, error } = removeResult.error;
+          logger.warn(
+            `Error updating task ${id}:${type} to mark as unrecognized during claim: ${error.message}`,
+            logMeta
+          );
+        }
+      }
+    } catch (err) {
+      logger.warn(`Error updating tasks to mark as unrecognized during claim: ${err}`, logMeta);
+    }
+  }
+
+  const message = `task claimer claimed: ${finalResults.length}; stale: ${staleTasks.size}; conflicts: ${conflicts}; missing: ${missingTasks.size}; updateErrors: ${bulkErrors}; removed: ${removedCount};`;
+  logger.debug(message, logMeta);
+
+  const finalResult = {
+    stats: {
+      tasksUpdated: finalResults.length,
+      tasksConflicted: conflicts,
+      tasksClaimed: finalResults.length,
+    },
+    docs: finalResults,
+    timing: stopTaskTimer(),
```

Comment:
It's weird that even after adding this code the health API polling stats don't work 🤔 

```
"polling": {
  "claim_duration": {
    "p50": null,
    "p90": null,
    "p95": null,
    "p99": null
  },
  "duration": { "p50": null, "p90": null, "p95": null, "p99": null },
  "claim_conflicts": {
    "p50": null,
    "p90": null,
    "p95": null,
    "p99": null
  },
  "claim_mismatches": {
    "p50": null,
    "p90": null,
    "p95": null,
    "p99": null
  },
  "result_frequency_percent_as_number": {
    "Failed": 0,
    "NoAvailableWorkers": 0,
    "NoTasksClaimed": 0,
    "RanOutOfCapacity": 0,
    "RunningAtCapacity": 0,
    "PoolFilled": 0
  },
  "persistence": { "recurring": 0, "non_recurring": 0 }
}
```

Reply:
This looks like it's because we're no longer emitting an initial `pollingDelay` metric. In `x-pack/plugins/task_manager/server/monitoring/task_run_statistics.ts`, we're using a `combineLatest` to combine 4 observables. `combineLatest` won't emit unless all 4 observables emit at least once. Because the `mget` task claimer never emits an initial `pollingDelay` stat, none of the 4 metrics are emitted.

You can add this to force an initial value for the `pollingDelay` observable.

```
--- a/x-pack/plugins/task_manager/server/monitoring/task_run_statistics.ts
+++ b/x-pack/plugins/task_manager/server/monitoring/task_run_statistics.ts
@@ -190,7 +190,8 @@ export function createTaskRunAggregator(
         (taskEvent: TaskLifecycleEvent) =>
           isTaskManagerStatEvent(taskEvent) && taskEvent.id === 'pollingDelay'
       ),
-      map(() => new Date().toISOString())
+      map(() => new Date().toISOString()),
+      startWith(new Date().toISOString())
     ),
     // get the average ratio of polled tasks by their persistency
     taskPollingLifecycle.events.pipe(
```

Reply:
🤦 good catch, we'll need something like this I guess.

Reply:
I'm applying Ying's suggestion - 🤞🏻

Reply:
updated in 45b652eb350b4327d2deb3ca11d8df2820a4afbc

Reply:
Tested and it now works, thanks!